### PR TITLE
Feat/scrollable

### PR DESCRIPTION
**Pull request by manyx**

### CODE REVIEW

File: core/src/components/cat-scrollable/cat-scrollable.tsx [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+        <div
+          ref={el => (this.scrollElement = el)}
+          class='scrollable-content'>
+          <slot></slot>
+        </div>
+      </Host>
+    );
+  }
+
+  private attachEmitter(from: 'top' | 'left' | 'right' | 'bottom', emitter: EventEmitter<void>, buffer: number) {
```
Comment (vardidzel) [inline core/src/components/cat-scrollable/cat-scrollable.tsx:115]: no return types for private methods.

Comment (vardidzel) [reply core/src/components/cat-scrollable/cat-scrollable.tsx:115]: Also, please avoid the use of the internal type/interfaces, move `'top' | 'left' | 'right' | 'bottom'` separately.

**Review by vardidzel — CHANGES_REQUESTED** [review]

File: core/src/components/cat-scrollable/cat-scrollable.tsx [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+  tag: 'cat-scrollable',
+  styleUrl: 'cat-scrollable.scss',
+  shadow: true
+})
+export class CatScrollable {
+  scrollElement?: HTMLElement;
+  scrollWrapperElement?: HTMLElement;
+  private readonly init = new Subject<void>();
+  private readonly destroyed = new Subject<void>();
+  private scrolled!: Observable<Event>;
```
Comment (fynnfeldpausch) [inline core/src/components/cat-scrollable/cat-scrollable.tsx:15]: get rid of the `!` here

**Review by fynnfeldpausch — CHANGES_REQUESTED** [review]

File: core/src/components/cat-scrollable/cat-scrollable.tsx (hunk 2 above)
Comment (manyx) [reply core/src/components/cat-scrollable/cat-scrollable.tsx:15]: it doesn't work without !

**Review by fynnfeldpausch — APPROVED** [review]

**Pull request merged**